hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/LegacyReplicationManager.java

```diff
@@ -517,6 +518,20 @@ protected void processContainer(ContainerInfo container,
           return;
         }
 
+        /*
+         * A QUASI_CLOSED container may have some UNHEALTHY replicas with the
+         * same Sequence ID as the container. RM should try to maintain one
+         * copy of such replicas when there are no healthy replicas that
+         * match the container's Sequence ID.
+         */
+        List<ContainerReplica> vulnerableUnhealthy =
+            replicaSet.getVulnerableUnhealthyReplicas(nodeManager);
+        if (!vulnerableUnhealthy.isEmpty()) {
+          handleVulnerableUnhealthyReplicas(replicaSet, vulnerableUnhealthy,
+              placementStatus);
+          return;
```

Comment:
Should such a container be marked as under replicated?

Reply:
You could argue either way, but perhaps it is better to make it under replicated, as it will potentially block decommission until it is copied, and as we need to make a copy we are saying the data is at risk, which is what under replication generally means.

Reply:
Makes sense. I've marked it as under replicated in the latest commit.